### PR TITLE
[Morningstar] Modernize

### PR DESCRIPTION
**Pull request by TRox1972**
## Please follow the guide below
- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your _pull request_ (like that [x])
- Use _Preview_ tab to see how your _pull request_ will actually look like

---
### Before submitting a _pull request_ make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)
### What is the purpose of your _pull request_?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---
### Description of your _pull request_ and other information

Simplifies metadata extraction for Morningstar.

### CODE REVIEW

File: youtube_dl/extractor/morningstar.py [hunk 1]
```diff
@@ -16,32 +14,27 @@ class MorningstarIE(InfoExtractor):
...
 
     def _real_extract(self, url):
-        mobj = re.match(self._VALID_URL, url)
-        video_id = mobj.group('id')
+        video_id = self._match_id(url)
 
         webpage = self._download_webpage(url, video_id)
-        title = self._html_search_regex(
-            r'<h1 id="titleLink">(.*?)</h1>', webpage, 'title')
+        title = get_element_by_id('titleLink', webpage)
```
Comment (yan12125) [inline youtube_dl/extractor/morningstar.py]: ```title``` is mandatory while ```get_element_by_id``` fails silently if nothing is found.

File: youtube_dl/extractor/morningstar.py [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
 from __future__ import unicode_literals
 
-import re
-
 from .common import InfoExtractor
+from ..utils import get_element_by_id
 
 
 class MorningstarIE(InfoExtractor):
-    IE_DESC = 'morningstar.com'
```
Comment (yan12125) [inline youtube_dl/extractor/morningstar.py]: Why remove it?

File: youtube_dl/extractor/morningstar.py (hunk 1 above)
Comment (TRox1972) [inline youtube_dl/extractor/morningstar.py]: Should we change `get_element_by_id` to handle `fatal=True` cases? Or should I just raise an `ExtractorError` whenever title extraction fails?

File: youtube_dl/extractor/morningstar.py (hunk 2 above)
Comment (TRox1972) [inline youtube_dl/extractor/morningstar.py]: Other extractors don't usually specify `IE_DESC` to include `.com`, and it isn't part of Morningstar's name.

Comment (yan12125) [inline youtube_dl/extractor/morningstar.py]: Some extractors use the domain name as ```IE_DESC```, so I guess it's OK.

Comment (TRox1972) [inline youtube_dl/extractor/morningstar.py]: OK. Reversed.

File: youtube_dl/extractor/morningstar.py (hunk 1 above)
Comment (TRox1972) [inline youtube_dl/extractor/morningstar.py]: fixed

File: youtube_dl/extractor/morningstar.py [hunk 3]
```diff
@@ -16,32 +18,30 @@ class MorningstarIE(InfoExtractor):
...
     def _real_extract(self, url):
-        mobj = re.match(self._VALID_URL, url)
-        video_id = mobj.group('id')
-
+        video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
-        title = self._html_search_regex(
-            r'<h1 id="titleLink">(.*?)</h1>', webpage, 'title')
+        
+        title = get_element_by_id('titleLink', webpage) or ''
```
Comment (yan12125) [inline youtube_dl/extractor/morningstar.py]: Is ```or ''``` necessary?

Comment (TRox1972) [inline youtube_dl/extractor/morningstar.py]: removed

File: youtube_dl/extractor/morningstar.py [hunk 4]
```diff
@@ -16,32 +18,30 @@ class MorningstarIE(InfoExtractor):
...
-        video_id = mobj.group('id')
-
+        video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
-        title = self._html_search_regex(
-            r'<h1 id="titleLink">(.*?)</h1>', webpage, 'title')
+        
+        title = get_element_by_id('titleLink', webpage)
+        if not title:
+            raise ExtractorError('Unable to extract title for %s.' % video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/morningstar.py]: This should be done in generic way similar to `_search_regex`.

**Pull request closed without merging**